enhancements/two-node-fencing/tnf.md

```diff
@@ -708,6 +666,42 @@ This collection of diagrams collects a series of scenarios where both nodes fail
 
 ![Diagrams of Multi-Node Failure Scenarios](etcd-flowchart-both-nodes-reboot-scenarios.svg)
 
+#### Status Propagation with PacemakerCluster Health Check
+An important goal of Two Node OpenShift with Fencing is ensuring an early warning when the cluster enters a state where automatic recovery from quorum loss is not possible. To provide this
+warning, we need pacemaker health information to be available in the cluster. An example of this would be if the cluster administrator rotated their BMC password without updating the fencing secret
+in the cluster. This would be caught by the pacemaker monitoring checks, but something in the cluster needs to propagate that information to the user directly.
+
+To acheive this, we plan on using two new controllers in CEO. The first is a status collector which syncs every 30 seconds to gather that current state of pacemaker via `sudo pcs status xml`.
+This is parsed to create a `PacemakerCluster` status object, a singleton resource created by CEO when the transition to etcd running externally is completed.
+Additionally, it creates events for the follow:
+- Errors events when kubelet, etcd, or the fencing agent on a node enter an unhealthy state
+- Warnings events when kubelet, etcd, or the fencing agent on a node are started or stopped
+- Warnings events fencing actions are taken by the cluster
+
+The `PacemakerCluster` resource provides the cluster with key information that CEO can use to determine the overall health and threats to etcd. It consists of:
+- A `lastUpdated` timestamp tracking when the status was last updated
+- Cluster-level conditions tracking overall health, maintenance mode, and expected node count
+- A list of `PacemakerClusterNodeStatus` objects representing the state of the nodes registered by pacemaker
+
+The `PacemakerClusterNodeStatus` consists of:
+- The name and IP address of the node
```

Comment:
Lets make sure the IP address API we build reflects that of the Node object, ie has the ability to provide multiple addresses and specify the type of the address